lib/3scale/backend/transactor.rb

```diff
@@ -70,12 +72,18 @@ def validate(oauth, provider_key, report_usage, params, request_info)
         # significant.
         params[:app_id] = nil if app_id && app_id.empty?
 
-        if oauth
-          raise ApplicationNotFound.new nil if app_id.nil?
-          validators = Validators::OAUTH_VALIDATORS
-        else
-          validators = Validators::VALIDATORS
-        end
+        # While OIDC without an app_id makes little sense, we would break existing
```

Comment:
I'm confused by this description. I think what it's saying does not match what the line below does.